docs/index.bs

```diff
@@ -3110,6 +3178,100 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
+          1. [=header list/Append=] to |preloadRequestHeaders| a new [=header=] whose [=header/name=] is \`<code>Service-Worker-Navigation-Preload</code>\` and [=header/value=] is |registration|'s [=navigation preload header value=].
+          1. Set |preloadRequest|'s [=service-workers mode=] to "`none`".
+          1. Let |preloadFetchController| be null.
+          1. Run the following substeps [=in parallel=], but [=abort when=] |controller|'s [=fetch controller/state=] is "<code>terminated</code>" or "<code>aborted</code>":
+              1. Set |preloadFetchController| to the result of [=Fetch|fetching=] |preloadRequest|.
+
+                  To [=fetch/processResponse=] for |navigationPreloadResponse|, run these substeps:
+
+                  1. If |navigationPreloadResponse|'s [=response/type=] is "`error`", reject |preloadResponse| with a `TypeError` and terminate these substeps.
+                  1. Associate |preloadResponseObject| with |navigationPreloadResponse|.
+                  1. Resolve |preloadResponse| with |preloadResponseObject|.
+          1. [=If aborted=], then:
+              1. Let |deserializedError| be the result of [=deserialize a serialized abort reason=] given null and |workerRealm|.
+              1. [=fetch controller/Abort=] |preloadFetchController| with |deserializedError|.
+      1. Else, resolve |preloadResponse| with undefined.
+      1. Return the result of [=Create Fetch Event and Dispatch=] with |request|, |registration|, |useHighResPerformanceTimers|, |timingInfo|, |workerRealm|, |reservedClient|, |preloadResponse|, and |raceResponse|.
+  </section>
+
+  <section algorithm>
+    <h3 id="create-fetch-event-and-dispatch-algorithm"><dfn>Create Fetch Event and Dispatch</dfn></h3>
+      : Input
+      :: |request|, a [=/request=]
+      :: |registration|, a [=/service worker registration=]
+      :: |useHighResPerformanceTimers|, a boolean
+      :: |timingInfo|, a [=service worker timing info=]
+      :: |workerRealm|, a [=relevant realm=] of the [=service worker/global object=]
+      :: |reservedClient|, a [=request/reserved client=]
+      :: |preloadResponse|, a [=promise=]
+      :: |raceResponse|, a [=race response=]
+      : Output
+      :: a [=/response=]
+
+      1. Let |response| be null.
+      1. Let |eventCanceled| be false.
+      1. Let |client| be |request|'s [=request/client=].
+      1. Let |activeWorker| be |registration|'s <a>active worker</a>.
+      1. Let |eventHandled| be null.
+      1. Let |handleFetchFailed| be false.
+      1. Let |respondWithEntered| be false.
+      1. Let |shouldSoftUpdate| be true if any of the following are true, and false otherwise:
+          * |request| is a [=non-subresource request=].
+          * |request| is a [=subresource request=] and |registration| is [=stale=].
+      1. Let |raceResponseMap| be |activeWorker|'s [=service worker/global object=]'s [=race response map=].
```

Comment:
this should move till after the run Service Worker step below as well to make sure global object has been initialized.

Reply:
Removed |raceResponseMap| at this point, and directly call _|activeWorker|'s [=service worker/global object=]'s [=race response map=]_ after the global object has been initialized.